lib/model.js

```diff
@@ -4788,8 +4789,8 @@ Model.compile = function compile(name, schema, collectionName, connection, base)
   applyStaticHooks(model, schema.s.hooks, schema.statics);
 
   model.schema = model.prototype.$__schema;
-  model.collection = model.prototype.collection;
-  model.$__collection = model.collection;
+  model.collection = collection;
+  model.$__collection = collection;
```

Comment:
i dont quite know why this is set here, but why not also set the symbol?

Reply:
`modelCollectionSymbol` is only set on `Model.prototype`, not `Model`. `modelCollectionSymbol` is necessary for naming conflicts in case both `collection` and `$collection` are schema paths. No risk of name conflicts with schema paths on Model though